lib/ansible/module_utils/basic.py

```diff
@@ -930,9 +930,10 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
 
     @property
     def tmpdir(self):
-        # if _ansible_tmpdir was not set, the module needs to create it and
-        # clean it up once finished.
-        if self._tmpdir is None:
+        # if _ansible_tmpdir was not set and we have a remote_tmp,
+        # the module needs to create it and clean it up once finished.
+        # otherwise it stays None and tempfile consumer will use system dir defaults
+        if self._tmpdir is None and self._remote_tmp:
```

Comment:
I think we need _remote_tmp here.  So we should either make sure we're always passing in a _remote_tmp or we should set a default here.  (Since we're already setting a default in action/__init__.py:_update_module_args, it feels like we should set this there as well).

Reply:
its stays None, which when consumed by tempfile will use system dirs, using default might not work as `~/.ansible/tmp` can break things and user changed it for a reason

Reply:
Hmmm... okay.   None is an implementation detail, though.  So what we probably want is if we aren't given a remote_tmp either, then we should create a tmpdir in the system tempdir, store that in self._tmpdir and return it from this function.

Reply:
so basically do the work tempfile already does?

Reply:
Yes.  That's the expectation if self._tmpdir is set to None when this property is accessed.

Reply:
This is still needed.